File: backend/src/main/java/com/vira/dply/mapper/UserMapper.java
package com.vira.dply.mapper;

import org.mapstruct.Mapper;

import com.vira.dply.dto.UserDto;
import com.vira.dply.entity.UserEntity;

@Mapper(componentModel = "spring")
public interface UserMapper {

    UserDto toDto(UserEntity entity);

    UserEntity toEntity(UserDto dto);
}

File: backend/src/main/kotlin/com/vira/dply/mapper/UserMapper.kt
package com.vira.dply.mapper

import com.vira.dply.dto.RegisterDto
import com.vira.dply.dto.UserDto
import com.vira.dply.model.User
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface UserMapper {

    fun toEntity(dto: RegisterDto): User

    fun toDto(entity: User): UserDto
}